src/test/java/org/broadinstitute/consent/http/service/DraftServiceTest.java

```diff
@@ -147,6 +148,19 @@ void testDeleteDraftsForUser() throws SQLException, IOException {
     draftService.deleteDraftsByUser(user2);
   }
 
+  @Test
+  void testAddAttachmentToDraft() throws SQLException, IOException {
+    when(gcsService.storeDocument(any(), anyString(), any())).thenReturn(
+        BlobId.of(UUID.randomUUID().toString(), UUID.randomUUID().toString()));
```

Comment:
This is OK but in general I think it's better to mock the layers directly beneath the one being tested.

In this case, instead of mocking `GCSService` and creating a `DraftFileStorageService`, you'd mock `DraftFileStorageService` and only mock `GCSService` when testing `DraftFileStorageService`.

Reply:
Thanks for your suggestion @pshapiro4broad.  I appreciate your perspective and think this begs a bit of a larger conversation about the scope of this projects tests in general (are they unit, are they integration, should we be running these tests without an underlying database, etc.) and will hold off making the change until we can have that discussion.